src/pages/index.js

```diff
@@ -3,82 +3,296 @@ import FormValidator from "../components/FormValidator.js";
+        name: formData.title,
+        job: formData.description,
+        avatar: user.getUserInfo().avatar,
+      });
+      editFormPopup.close();
+    })
+    .catch((error) => {
+      console.error("Failed to update user information:", error);
+    })
+    .finally(() => {
+      variables.editModalProfileSaveButton.textContent = "Save";
+    });
 });
 
+/* -------------------------------------------------------------------------- */
+/*                   Edit Button Event Listener                               */
+/* -------------------------------------------------------------------------- */
+
 variables.profileEditButton.addEventListener("click", () => {
   const userInfo = user.getUserInfo();
   variables.profileModalNameInput.value = userInfo.name.trim();
   variables.profileModalDescriptionInput.value = userInfo.job.trim();
   editFormPopup.open();
 });
 
-const addFormPopup = new PopupWithForm(
+editFormPopup.setEventListeners();
+
+/* -------------------------------------------------------------------------- */
+/*                    Get the initial cards from server                       */
+/* -------------------------------------------------------------------------- */
+
+getCards();
+
+let cardSection;
+
+function getCards() {
+  api
+    .getInitialCards()
+    .then((cards) => {
+      cardSection = new Section(
+        {
+          items: cards,
+          renderer: (card) => {
+            const cardElement = createCard(card);
+            cardSection.addItem(cardElement);
+          },
+        },
+        "#el-card-list"
+      );
+      cardSection.renderCards(cards);
+    })
+    .catch((error) => {
+      console.error("Failed to fetch cards:", error);
+    });
+}
+
+/* -------------------------------------------------------------------------------------------- */
+/*      Create new card and store the info in the server, then call to render card object       */
+/* -------------------------------------------------------------------------------------------- */
+
+function createNewCard(card) {
+  variables.createCardButton.textContent = "Creating...";
+  api
+    .createCard(card)
+    .then((cardData) => {
+      cardSection.addItem(createCard(cardData));
+      addCardPopup.close();
+      variables.addModalForm.reset();
+    })
+    .catch((error) => {
+      console.error("Failed to create new card", error);
+    })
+    .finally(() => {
+      variables.createCardButton.textContent = "Create";
+    });
+}
+
+/* -------------------------------------------------------------------------- */
+/*               Call API to submit card info to be created                   */
+/* -------------------------------------------------------------------------- */
+
+const addCardPopup = new PopupWithForm(
   "#profile-add-modal",
   ({ title: name, link }) => {
-    renderCard({ name, link });
-    variables.addModalForm.reset();
+    createNewCard({ name, link });
     addModalFormValidator.disableSubmitButton();
```

Comment:
You need to `disable` the button only after **resetting** the form when you have submitted a card in block `then` to be able to click it again if there is a server error